lib/src/domain/common/openalias.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:basic_utils/basic_utils.dart';
+
+String formatDomainName(String name) {
+  String formattedName = name;
+
+  if (name.contains(".")) {
+    formattedName = name.replaceAll("@", ".");
+  }
+
+  return formattedName;
+}
+
+Future<Map<String, String>> fetchXmrAddressAndRecipientName(String name) async {
+  final map = {"recipient_address" : name, "recipient_name" : name};
+
+  try {
+    await DnsUtils.lookupRecord(name, RRecordType.TXT, dnssec: true).then((txtRecord) {
```

Comment:
Use await or promises (.then()) don't mix it.

Reply:
For example `final record = await DnsUtils.lookupRecord(...);` and then use record "synchronized".